src/tactic/omega/clause.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2019 Seul Baek. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Seul Baek
+
+(([t₁,...tₘ],[s₁,...,sₙ]) : clause) encodes the constraints
+0 = ⟦t₁⟧ ∧ ... ∧ 0 = ⟦tₘ⟧ ∧ 0 ≤ ⟦s₁⟧ ∧ ... ∧ 0 ≤ ⟦sₙ⟧, where
+⟦t⟧ is the value of (t : term).
+-/
+
+import tactic.omega.term
+
+namespace omega
+
+@[reducible] def clause := (list term) × (list term)
```

Comment:
I assume this is the reason for all the list.get and list.set stuff earlier. Why don't you use `array`, or `buffer`? These types already have builtin functions for read and write, and they are faster because they are actually array reads and writes.

Reply:
Are clauses used in proof terms or in proof search?

Reply:
Looking at the profiling results, by far the largest portion of execution time is spent on _reflecting_ lists to exprs, not accessing individual members of lists or performing list operations to obtain other lists. Would the use of arrays or buffers shorten the time spent on this reflection?

Reply:
I suspect it would increase it, but I don't know for sure. Arrays will make for long exprs, and might not compute well in the kernel, depending on how you reflect them. It could be optimal to use arrays in the VM and reflect them to lists, but this is probably an extra level of complication.

Reply:
The time spent in Fourier-Motzkin elimination with the VM is negligible, somewhere around 3-5% of total. I guess I'll stick with lists then.